tests/runner.py

```diff
@@ -922,6 +922,8 @@ def expect_fail(self, cmd, **args):
     Return the stderr of the subprocess.
     """
     proc = run_process(cmd, check=False, stderr=PIPE, **args)
+    if proc.returncode == 0:
```

Comment:
is this for debugging? then shouldn't it be `!=`? Or maybe I don't understand the motivation here.

Reply:
yes this is for debugging failures.

This is `expect_fail` so we expect that return code to be non-zero.   If its *is* zero then we don't want to sliently swallow the stderr (like we are currently doing).

Reply:
Hmm. I can think of a nicer way maybe..